homework-submission/homework-week1/app.js

```diff
@@ -0,0 +1,160 @@
+const Item = ({ id, description, deadline, done, removeItem }) => {
+  return (
+    <li key={id} className={done === true ? 'true' : ''}>
+      {description} {deadline} {done}
+      <span className="delete" onClick={() => removeItem(id)}>
+        DELETE
+      </span>
+    </li>
+  );
+};
+
+const TodoItems = ({ toDoList, removeItem }) => {
+  const length = toDoList.length;
+  const ListItems = length ? (
+    toDoList.map(({ id, done, description, deadline }) => {
+      return (
+        <Item
+          key={id}
+          id={id}
+          description={description}
+          deadline={deadline}
+          done={done}
+          removeItem={removeItem}
+        />
+      );
+    })
+  ) : (
+    <p>There is no items</p>
+  );
+
+  return <ul className="todo-items">{ListItems}</ul>;
+};
+
+class AddItems extends React.Component {
+  state = {
+    description: '',
+    deadline: '',
+  };
+
+  handelChange = e => {
+    this.setState({
+      [e.target.id]: e.target.value,
+    });
+  };
+
+  error() {
+    let element = document.getElementById('error');
+    element.classList.remove('hidde');
+  }
+
+  handelSubmit = e => {
+    e.preventDefault();
+    if (e.target['description'].value !== '' && e.target['deadline'].value !== '') {
+      this.props.addItem(this.state);
+      this.setState({
+        description: '',
+        deadline: '',
+      });
+    } else {
+      this.error();
+    }
+  };
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handelSubmit}>
+          <label>Description: </label>
+          <input
+            type="text"
+            id="description"
+            value={this.state.description}
+            placeholder="Enter a short description here..."
+            onChange={this.handelChange}
+          />
+
+          <label>Deadline: </label>
+          <input
+            type="date"
+            id="deadline"
+            placeholder="Enter deadline..."
+            value={this.state.deadline}
+            onChange={this.handelChange}
+          />
+          <input type="submit" value="Add" className="add" />
+        </form>
+      </div>
+    );
+  }
+}
+
+class App extends React.Component {
+  state = {
+    toDoList: [
+      {
+        id: 1,
+        description: 'Get out of bed',
+        deadline: '2017-09-11',
+        done: true,
+      },
+      {
+        id: 2,
+
+        description: 'Brush teeth',
+        deadline: '2017-09-10',
+        done: false,
+      },
+      {
+        id: 3,
+        description: 'Eat breakfast',
+        deadline: '2017-09-09',
+        done: false,
+      },
+    ],
+  };
+
+  removeItem = id => {
+    const toDoList = this.state.toDoList.filter(item => {
+      return item.id !== id;
+    });
+    this.setState({ toDoList });
+  };
+
+  addItem = item => {
+    const random = Math.random(Math.random() * 9999);
```

Comment:
This will solve the problem, but why a random inside a random? 😕

Reply:
It was my fault,  I would say random inside round.